src/main/java/com/sugarcrm/candybean/webservices/WS.java

```diff
@@ -5,232 +5,251 @@
+	}
+
+	/**
+	 * Send a DELETE, GET, POST, or PUT http request
+	 * @param op The type of http request
+	 * @param uri The http endpoint
+	 * @param payload Map of key value pairs for the body
+	 * @param postHeaders List of Maps of header key value pairs
+	 * @param body String representation of the request body (ignored)
+	 * @return Key Value pairs of the response
+	 * @throws Exception When http request failed
+	 * @deprecated This is a work around for old compatibility, use {@link #request(OP, String, Map, String, ContentType)}
+	 * or {@link #request(OP, String, Map, Map)}
+	 */
+	@Deprecated
+	public static Map<String, Object> request(OP op, String uri, Map<String, String> payload, String body, ArrayList<HashMap<String, String>> postHeaders) throws Exception {
+		HashMap<String, String> headers = new HashMap<>();
+		for (HashMap<String, String> map : postHeaders) {
+			for (Map.Entry<String, String> entry : map.entrySet()) {
+				headers.put(entry.getKey(), entry.getValue());
+			}
+		}
+		HashMap<String,Object> newPayload = new HashMap<>();
+		for (Map.Entry<String,String> entry: payload.entrySet()) {
+			newPayload.put(entry.getKey(), entry.getValue());
+		}
+
+		if (body == null) {
+			return request(op, uri, headers, newPayload);
+		} else {
+			return request(op, uri, headers, body, ContentType.DEFAULT_TEXT);
 		}
-		return value;
 	}
-	
-	public static Map<String, Object> request(OP op, String uri, Map<String, String> headers, String body) throws Exception {
-		return request(op, uri, headers, body, null);
+
+	/**
+	 * Send a DELETE, GET, POST, or PUT http request using a JSON body
+	 * @param op The type of http request
+	 * @param uri The http endpoint
+	 * @param headers Map of header key value pairs
+	 * @param body Map of Key Value pairs
+	 * @return Key Value pairs of the response
+	 * @throws Exception If HTTP request failed
+	 */
+	public static Map<String, Object> request(OP op, String uri, Map<String, String> headers, Map<String, Object> body) throws Exception {
+		return request(op, uri, headers, body, ContentType.APPLICATION_JSON);
 	}
-	
-	public static Map<String, Object> request(OP op, String uri, Map<String, String> headers, String body, ArrayList<HashMap<String, String>> postHeaders) throws Exception {		
-		HttpUriRequest request = null;
-	    Map<String, Object> mapParse = null;
+
+	/**
+	 * Send a DELETE, GET, POST, or PUT http request, intended for multipart data and JSON requests
+	 * @param op The type of http request
+	 * @param uri The http endpoint
+	 * @param headers Map of header key value pairs
+	 * @param body Map of Key Value pairs
+	 * @param contentType The intended content type of the body
+	 * @return Key Value pairs of the response
+	 * @throws Exception If HTTP request failed
+	 */
+	public static Map<String, Object> request(OP op, String uri, Map<String, String> headers, Map<String, Object> body, ContentType contentType) throws Exception {
```

Comment:
Let's write some documentation about what are the types of Map is accepted for body.